### DISCUSSION TITLE
About label encoding in the MIL tutorial

### DISCUSSION THREAD

**Discussion by mdrio** [Project-MONAI/MONAI#7699] [Q&A] [answered]
Hi, I am wondering why the labels in the [MIL tutorial](https://github.com/Project-MONAI/tutorials/blob/main/pathology/multiple_instance_learning/panda_mil_train_evaluate_pytorch_gpu.py), implemented in the class LabelEncodeIntegerGraded, have  1 filled in up to label index and is not one-hot.  If I understand well, label 2 will be encoded in 11000, while in one-hot fashion should be 01000. 
I also wonder if the same MIL approach can be used for a multi label learning.

**Comment by KumoLiu** [comment, marked as answer]
Hi @mdrio,

In this Multiple Instance Learning (MIL) tutorial, labels are encoded using a graded label encoding instead of a traditional one-hot encoding. This approach is taken because, in this context, the class levels are ordinal i.e., class 1 is lower than class 2, class 2 is lower than class 3, and so on. Therefore, it is understood that class 2 combines information from both class 1 and itself, hence it's encoded as 11000 instead of 01000.
In this type of encoding, a class i would be represented by i ones followed by zeros. This encoding effectively represents this grading context where higher classes encompass more lower-class information.
Regarding your second question, a MIL approach can indeed be adapted to multi-label learning. In a multi-label setting, each instance could be assigned multiple labels, while in MIL, all instances in a bag share a common label. Although there are differences in these settings, the overall idea of learning a useful representation from a set of instances is similar. Therefore, the MIL paradigm can be adjusted to handle instances having multiple labels.

Hope it helps, thanks.
